### PR TITLE
fix(sql): remove sql skip-adding-user question

### PR DESCRIPTION
**Pull request by xzf0587**
1. remove sql skip-adding-user question.
2. set default skip-adding-user value based on azureAccountProvider?.getIdentityCredentialAsync

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/sql/plugin.ts [hunk 1]
```diff
@@ -134,14 +98,15 @@ export class SqlPluginImpl {
...
-      const skipAddingUser = ctx.answers![Constants.questionKey.skipAddingUser] as string;
-      if (skipAddingUser) {
-        this.config.skipAddingUser = skipAddingUser === "true" ? true : false;
-        ctx.config.set(Constants.skipAddingUser, this.config.skipAddingUser);
-      }
+    const skipAddingUser = ctx.config.get(Constants.skipAddingUser);
+    if (skipAddingUser === undefined) {
+      this.config.skipAddingUser = (await ctx.azureAccountProvider?.getIdentityCredentialAsync())
+        ? false
+        : true;
```
Comment (Alive-Fish) [inline packages/fx-core/src/plugins/resource/sql/plugin.ts:105]: Is that the opposite?

**Comment by Alive-Fish** [comment]
I think it's OK for the most cases, but some corner cases may have problems, such as provisioning in CLI first and then provision in VSCode extension.

**Review by xzf0587 — COMMENTED** [review]
> I think it's OK for the most cases, but some corner cases may have problems, such as provisioning in CLI first and then provision in VSCode extension.

If developers haven't modify the skip-adding-user config in env file, leave it as default. The project will work.
Only  specifying the skip-adding-user true in env file will make the cli provision failed.

**Comment by wenytang-ms** [comment]
it's better merge as a hotfix not a feature

**Comment by adashen** [comment]
> > I think it's OK for the most cases, but some corner cases may have problems, such as provisioning in CLI first and then provision in VSCode extension.
> 
> If developers haven't modify the skip-adding-user config in env file, leave it as default. The project will work.
> Only specifying the skip-adding-user false in env file will make the cli provision failed.

Let's add both test cases and do the verification

**Comment by xzf0587** [comment]
> > > I think it's OK for the most cases, but some corner cases may have problems, such as provisioning in CLI first and then provision in VSCode extension.
> > 
> > 
> > If developers haven't modify the skip-adding-user config in env file, leave it as default. The project will work.
> > Only specifying the skip-adding-user false in env file will make the cli provision failed.
> 
> Let's add both test cases and do the verification

Added the uint test for skip-adding-user.
The e2e test does not support test cli scenario which the identity credential is undefined.

**Review by adashen — APPROVED** [review]

**Review by Alive-Fish — APPROVED** [review]
look good to me

File: packages/fx-core/src/plugins/resource/sql/plugin.ts [hunk 2]
```diff
@@ -134,14 +124,15 @@ export class SqlPluginImpl {
...
-    this.config.skipAddingUser = ctx.config.get(Constants.skipAddingUser) as boolean;
-    if (ctx.answers?.platform === Platform.CLI) {
-      const skipAddingUser = ctx.answers![Constants.questionKey.skipAddingUser] as string;
-      if (skipAddingUser) {
-        this.config.skipAddingUser = skipAddingUser === "true" ? true : false;
-        ctx.config.set(Constants.skipAddingUser, this.config.skipAddingUser);
-      }
+    const skipAddingUser = ctx.config.get(Constants.skipAddingUser);
+    if (skipAddingUser === undefined) {
+      this.config.skipAddingUser = (await ctx.azureAccountProvider?.getIdentityCredentialAsync())
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/resource/sql/plugin.ts:129]: Can this be simplified to :
```
this.config.skipAddingUser = !(await ctx.azureAccountProvider?.getIdentityCredentialAsync())
```
